pkg/controller/monitor/monitor_controller.go

```diff
@@ -265,6 +294,64 @@ func (r *ReconcileMonitor) Reconcile(ctx context.Context, request reconcile.Requ
 		return reconcile.Result{}, err
 	}
 
+	managementClusterConnection, err := utils.GetManagementClusterConnection(ctx, r.client)
+	if err != nil {
+		log.Error(err, "Error reading ManagementClusterConnection")
+		r.status.SetDegraded("Error reading ManagementClusterConnection", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	// In managed clusters with certificate management disabled, successful reconciliation of non-NetworkPolicy resources
+	// in the monitor controller ensures that NetworkPolicy is reconcilable (by enabling* the creation of containing Tier
+	// and License). Therefore, to prevent a chicken-and-egg scenario, we only reconcile non-NetworkPolicy resources once
+	// we can confirm that all requirements to reconcile NetworkPolicy have been met.
+	//
+	// * In managed clusters with certificate management disabled, the License can only be pushed once Guardian is deployed,
+	//   which depends on the Prometheus TLS secret created by monitor. As always, the Tier can only be created once the
+	//   License is available.
+	includeV3NetworkPolicy := true
+	monitorReconcilesV3NetworkPolicyRequirements := managementClusterConnection != nil && !certificateManager.KeyPair().UseCertificateManagement()
+	if monitorReconcilesV3NetworkPolicyRequirements {
+		// Only wait for and render NetworkPolicy once all NetworkPolicy requirements are met.
+		// Do not degrade if tier has not been created.
+		if utils.IsV3NetworkPolicyReconcilable(ctx, r.client, networkpolicy.TigeraComponentTierName) {
+			if !r.tierWatchReady.IsReady() {
+				r.status.SetDegraded("Waiting for Tier watch to be established", "")
+				return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+			}
+
+			if !r.policyWatchesReady.IsReady() {
+				r.status.SetDegraded("Waiting for NetworkPolicy watches to be established", "")
+				return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+			}
+		} else {
+			includeV3NetworkPolicy = false
+		}
+	} else {
+		// Wait for and render NetworkPolicy as usual.
```

Comment:
I think we could restructure this branching to avoid needing to write out this section twice. Couldn't we do something like this?


```go
// No matter what, we wait for watches to be established.
if ! r.tierWatchReady() {
  return
}

if !policyWatchReady() {
  return
}

// Determine if we should include policy.
includePolicy = true
if monitorReconcilesV3NetworkPolicy && !utils.IsReconcilable {
  includePolicy = false
}

if includePolicy {
  // Check that the allow-tigera tier exists. 
}
```

Reply:
As discussed, tier and policy watches need not block controllers from reconciling if they are not ready. We'll rely on the tier being available as a strong enough signal that we can proceed.